app/lib/origin_page.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class OriginPage
+  include Rails.application.routes.url_helpers
+  include TraineeHelper
+  include Breadcrumbable
```

Comment:
I'm thinking out loud here, but I'm not quite sure what the distinction between `OriginPage` and `Breadcrumbable` should be.

At the moment `Breadcrumbable` is responsible for saving, and `OriginPage` is responsible for fetching... But `OriginPage` also has to include `Breadcrumbable` since that's where all the helper methods live 🤔 

Is the distinction between the two correct? Should `OriginPage` actually include all the helper methods and implement a public `save!` as well as `path`?

```
module Breadcrumbable
  def save_origin_page_for(trainee)
    OriginPage.new(trainee, session, request).save!
  end
end
```

I'm not sure what my view is though 😂   happy to keep as is for now :)

Reply:
We also had similar thoughts but didn't move forward on it. I think we should move all the action into the `OriginPage` do what you suggested